aws/resource_aws_s3_bucket.go

```diff
@@ -76,6 +76,14 @@ func resourceAwsS3Bucket() *schema.Resource {
 				Default:       "private",
 				Optional:      true,
 				ConflictsWith: []string{"grant"},
+				ValidateFunc: validation.StringInSlice([]string{
```

Comment:
Instead of hardcoding them, how about just using `BucketCannedACL_Values()`?

Reply:
I dont really have a preference. But, looking at other validation functions, they seem to list out all the values. I dont think I want to go against the style?

https://github.com/terraform-providers/terraform-provider-aws/blob/0544ae1fe0f788616cc483df8d0e68d3dd36a0ed/aws/resource_aws_s3_bucket.go#L120
https://github.com/terraform-providers/terraform-provider-aws/blob/0544ae1fe0f788616cc483df8d0e68d3dd36a0ed/aws/resource_aws_s3_bucket.go#L406
https://github.com/terraform-providers/terraform-provider-aws/blob/0544ae1fe0f788616cc483df8d0e68d3dd36a0ed/aws/resource_aws_s3_bucket.go#L453

Reply:
Hi @NikolaeVarius. We've actually been changing over to the `_Values()` functions so that new values are automatically picked up when we update the AWS SDK. We're tracking this in #14601

The `_Values()` functions were actually added by one of our maintainers in https://github.com/aws/aws-sdk-go/pull/3447 🙂